### ISSUE TITLE
Unmarshalling to a Julia type when type has Union Values?

### ISSUE THREAD

**Issue by MrDataPsycho** [lwabeke/Unmarshal.jl#31]
Hi,
I am trying to unmarshal a json to a Struct type where there is Union in type attributes but it does not work. Is there a way to do that?

That function works:
```
input = "{ \"id\": 25073877, \"id_str\": \"25073877\"}"

struct UserX
    id::Int64
    id_str::String
end

Unmarshal.unmarshal(UserX, JSON.parse(input))
```

But Having Union in the struct Marshalling is not working:

```
struct UserY
    id::Union{Int64, Nothing}
    id_str::Union{String, Nothing}
end

Unmarshal.unmarshal(UserY, JSON.parse(input))
```

this through following error:

```
ERROR: MethodError: unmarshal(::Type{Union{Nothing, String}}, ::String, ::Bool, ::Int64) is ambiguous. Candidates:
  unmarshal(DT::Type, parsedJson::String, verbose::Bool, verboseLvl::Int64) in Unmarshal at /home/datapsycho/.julia/packages/Unmarshal/G72dP/src/Unmarshal.jl:38
  unmarshal(::Type{Union{Nothing, T}}, x, verbose::Bool, verboseLvl::Int64) where T in Unmarshal at /home/datapsycho/.julia/packages/Unmarshal/G72dP/src/Unmarshal.jl:232
Possible fix, define
  unmarshal(::Type{Union{Nothing, T}}, ::String, ::Bool, ::Int64) where T
```

Is it possible to unmarshal such data?

**Comment by lwabeke** [comment]
Hi.

```   id_str::Union{Int64, Nothing}``` should already work.
The problem was with ```   id_str::Union{String, Nothing}```. I had a function that allowed parsing of strings i.e. 
``` Unmarshal.unmarshal(String, JSON.parse(json("Test"))) == "Test" ``` which was causing the problems. I tweaked that function and it seems to work for your test case now. It should be fixed in either master or then v0.4.1 as soon as that release has propogated through the release mechanisms.

**Comment by MrDataPsycho** [comment]
Hi @iwabeke,
I pulled form latest master branch and it works. Thanks for such quick response. Closing the issue.
Thanks
